pygmt/sampling.py

```diff
@@ -0,0 +1,78 @@
+"""
+GMT modules for Sampling of 1-D and 2-D Data
+"""
+import pandas as pd
+import xarray as xr
+
+from .clib import Session
+from .helpers import (
+    build_arg_string,
+    fmt_docstring,
+    GMTTempFile,
+    data_kind,
+    dummy_context,
+)
+from .exceptions import GMTInvalidInput
+
+
+@fmt_docstring
+def grdtrack(table: pd.DataFrame, grid: xr.DataArray, newcolname: str = "z_", **kwargs):
```

Comment:
I'm still not sure about type hints. Do you have a particular use case for these?

Reply:
Kind of slipped them in out of habit. I find it useful to see at a glance what type of inputs to provide right at the very first line of the function. All that without looking at the docstring, and I would argue it's a more noticeable place down the line for maintenance reasons (if the API changes). Also, some IDEs like PyCharm use it to [validate inputs](https://www.jetbrains.com/help/pycharm/type-hinting-in-product.html#validate-type-hints). That said, I can remove it for consistency with the rest of PyGMT's codebase :)

Reply:
The input `table` could be a file, a pd.DataFrame or a numpy array. Does the type hint really make sense?

Reply:
Actually it *should* be a pandas.DataFrame as [these lines](https://github.com/GenericMappingTools/pygmt/blob/3776214536a2433b348bb948f56434150dd44d44/pygmt/sampling.py#L84-L85) will access the table's `column` attribute and use that for the output table.

The `grid` though does work with either a file or xarray.DataArray input though. In which case, we could specify multiple 'options' (see e.g. [this stackoverflow question](https://stackoverflow.com/questions/34647966/how-to-express-multiple-types-for-a-single-parameter-or-a-return-value-in-docstr)).

Reply:
Actually it *should* be a pandas.DataFrame as [these lines](https://github.com/GenericMappingTools/pygmt/blob/3776214536a2433b348bb948f56434150dd44d44/pygmt/sampling.py#L84-L85) will access the table's `column` attribute and use that for the output table.

The `grid` though does work with either a file or xarray.DataArray input though. In which case, we could specify multiple 'options' (see e.g. [this stackoverflow question](https://stackoverflow.com/questions/34647966/how-to-express-multiple-types-for-a-single-parameter-or-a-return-value-in-docstr)).

Reply:
I mean `grdtrack` should also accept a filename as its input.

Reply:
Accepting an arbitrary points file for grdtrack complicates things quite a bit, but I'll work on it.